eng/Packages.Data.props

```diff
@@ -171,7 +171,7 @@
     All should have PrivateAssets="All" set so they don't become package dependencies
   -->
   <ItemGroup>
-    <PackageReference Update="Microsoft.Azure.AutoRest.CSharp" Version="3.0.0-beta.20230607.3" PrivateAssets="All" />
```

Comment:
why the update here? lets try not to update files thats not part of this feature